app/services/sandbox_url_updator.rb

```diff
@@ -23,9 +26,73 @@ def update
     end
   end
 
+  private
+
+  def process_input
```

Comment:
I don't like how much custom code for username validation is happening here, especially since there is no guarantee that it will avoid problems and result in a valid sandbox URL.

I better strategy might be to find the user by username; if the user exists, that will definitely be a valid sandbox URL, and if the user does not exist, we should disallow using that sandbox anyway (even if it's technically a usable URL).